pkg/block/metadata/meta.go

```diff
@@ -53,6 +53,8 @@ type Meta struct {
 type Thanos struct {
 	Labels     map[string]string `json:"labels"`
 	Downsample ThanosDownsample  `json:"downsample"`
+	// Size of the block in bytes of the original block (before compaction and/or 1st upload).
```

Comment:
This semantic is quite odd. What do you mean before compaction? Meta JSON is always bound to the block and block is immutable, so I think this detail is not necessary. Meta.json should state size in bytes related to the block is states metadata for (: 

Also curious, can we make it separate `indexSizeBytes`  and `chunkTotalSizeBytes`? Those are the biggest blocks that matter. This info will enormously help in knowing the file sizes in advance (: for other certain bucket operations.

I think I like it. Alternative is change bucket API to fetch some metadata (size), but this sounds like some easy way to achieve this.

Reply:
Actually, let me invest a bit more time into seeing if we could add a new method to our remote object storage interface which would return the size of a specified object. Perhaps I will split it into a separate PR (if it's feasible to make it at all).